hasher-matcher-actioner/hmalib/lambdas/fetcher.py

```diff
@@ -167,33 +172,32 @@ def lambda_handler(event, context):
             supported_signal_types=[VideoMD5Signal, PdqSignal],
         )
 
-        indicator_store.load_checkpoint()
+        try:
+            indicator_store.load_checkpoint()
 
-        if indicator_store.stale:
-            logger.warning(
-                "Store for %s - %d stale! Resetting.",
-                collab.privacy_group_name,
-                int(collab.privacy_group_id),
-            )
-            indicator_store.reset()
+            if indicator_store.stale:
+                logger.warning(
+                    "Store for %s - %d stale! Resetting.",
+                    collab.privacy_group_name,
+                    int(collab.privacy_group_id),
+                )
+                indicator_store.reset()
 
-        if indicator_store.fetch_checkpoint >= now.timestamp():
-            continue
+            if indicator_store.fetch_checkpoint >= now.timestamp():
+                continue
 
-        delta = indicator_store.next_delta
+            delta = indicator_store.next_delta
 
-        try:
             delta.incremental_sync_from_threatexchange(
                 api, limit=MAX_DESCRIPTORS_UPDATED, progress_fn=ProgressLogger()
             )
-        except Exception as e:
+        except Exception:  # pylint: disable=broad-except
             # Don't need to call .exception() here because we're just re-raising
```

Comment:
nit/ignorable: it could be helpful to log the exception object as well and just not raise it along with the error message below

Reply:
great! (sanity check: and not trigger a retry?)